test/views/declaration/ExporterEoriNumberViewSpec.scala

```diff
@@ -0,0 +1,143 @@
+          .getElementById("Yes")
+          .getElementsByAttribute("checked")
+          .attr("value") mustBe YesNoAnswers.yes
+      }
+
+      "have proper messages for labels" in {
+        messages must haveTranslationFor("declaration.exporterEori.title")
+        messages must haveTranslationFor("declaration.exporterEori.eori.label")
+        messages must haveTranslationFor("declaration.exporterEori.hasEori.empty")
+        messages must haveTranslationFor("declaration.exporterEori.help-item1")
+      }
+
+      "display page title" in {
+        view.getElementsByClass(Styles.gdsPageLegend) must containMessageForElements("declaration.exporterEori.title")
+      }
+
+      "display section header" in {
+        view.getElementById("section-header") must containMessage("declaration.section.2")
+      }
+
+      "display eori question" in {
+        view.getElementsByClass("govuk-label--m") must containMessageForElements("declaration.exporterEori.eori.label")
+        view.getElementById("eori").attr("value") mustBe empty
+      }
+
+      "display 'Back' button that links to 'Is Declarant the Exporter' page" in {
+
+        val backButton = view.getElementById("back-link")
+
+        backButton must containMessage("site.back")
+        backButton.getElementById("back-link") must haveHref(controllers.declaration.routes.DeclarantExporterController.displayPage(Mode.Normal))
+      }
+
+      "display 'Save and continue' button" in {
+        val saveButton = view.getElementById("submit")
+        saveButton must containMessage("site.save_and_continue")
+      }
+
+      "display 'Save and return' button on page" in {
+        val saveAndReturnButton = view.getElementById("submit_and_return")
+        saveAndReturnButton must containMessage("site.save_and_come_back_later")
+      }
+
+      "handle invalid input" should {
+        "display errors when all inputs are incorrect" in {
+          val data = ExporterEoriNumber(Some(Eori("123456789")), YesNoAnswers.yes)
+          val form = ExporterEoriNumber.form().fillAndValidate(data)
+          val view = createView(form = form)
+
+          view must haveGovukGlobalErrorSummary
+          view must containErrorElementWithTagAndHref("a", "#eori")
+          view must containErrorElementWithMessageKey("declaration.eori.error.format")
+        }
+
+        "display errors when eori contains special characters" in {
+          val data = ExporterEoriNumber(eori = Some(Eori("12#$%^78")), hasEori = YesNoAnswers.yes)
+          val form = ExporterEoriNumber.form().fillAndValidate(data)
+          val view = createView(form = form)
+
+          view must haveGovukGlobalErrorSummary
+          view must containErrorElementWithTagAndHref("a", "#eori")
+          view must containErrorElementWithMessageKey("declaration.eori.error.format")
+        }
+      }
+    }
+
+    onJourney(CLEARANCE) { implicit request =>
+      "display 'Back' button that links to 'Person Presenting Goods Details ' page" in {
```

Comment:
Is this case dependent not only on the journey type but also on cached data? It is slightly misleading because there is a test for 'Back' button earlier and it includes Clearance as well.
I'm just thinking if it would be easier to read if the tests were structured in a different way...

Reply:
Done